src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+				superScrollView = superView;
+				break;
+			}
+
+			superView = superView.FindResponder<UIScrollView>();
+		}
+
+		// This is the case when the keyboard is already showing and we click another editor/entry
+		if (LastScrollView is not null)
+		{
+			// if there is not a current superScrollView, restore LastScrollView
+			if (superScrollView is null)
+			{
+				if (LastScrollView.ContentInset != StartingContentInsets)
+				{
+					AnimateScroll(() =>
+					{
+						LastScrollView.ContentInset = StartingContentInsets;
+						LastScrollView.ScrollIndicatorInsets = StartingScrollIndicatorInsets;
+					});
+				}
+
+				if (!LastScrollView.ContentOffset.Equals(StartingContentOffset))
+				{
+					if (View.FindResponder<UIStackView>() is UIStackView)
+						LastScrollView.SetContentOffset(StartingContentOffset, UIView.AnimationsEnabled);
+					else
+						LastScrollView.ContentOffset = StartingContentOffset;
+				}
+
+				StartingContentInsets = new UIEdgeInsets();
+				StartingScrollIndicatorInsets = new UIEdgeInsets();
+				StartingContentOffset = new CGPoint(0, 0);
+				LastScrollView = null;
+			}
+
+			// if we have different LastScrollView and superScrollViews, set the LastScrollView to the original frame
+			// and set the LastScrollView as the superScrolView
+			else if (superScrollView != LastScrollView)
+			{
+				if (LastScrollView.ContentInset != StartingContentInsets)
+				{
+					AnimateScroll(() =>
+					{
+						LastScrollView.ContentInset = StartingContentInsets;
+						LastScrollView.ScrollIndicatorInsets = StartingScrollIndicatorInsets;
+					});
+				}
+
+				if (!LastScrollView.ContentOffset.Equals(StartingContentOffset))
+				{
+					if (View.FindResponder<UIStackView>() is not null)
+						LastScrollView.SetContentOffset(StartingContentOffset, UIView.AnimationsEnabled);
+					else
+						LastScrollView.ContentOffset = StartingContentOffset;
+				}
+
+				LastScrollView = superScrollView;
+				if (superScrollView is not null)
+				{
+					StartingContentInsets = superScrollView.ContentInset;
+					StartingContentOffset = superScrollView.ContentOffset;
+
+					if (OperatingSystem.IsIOSVersionAtLeast(11, 1))
+						StartingScrollIndicatorInsets = superScrollView.VerticalScrollIndicatorInsets;
+					else
+						StartingScrollIndicatorInsets = superScrollView.ScrollIndicatorInsets;
+				}
+			}
+		}
+
+		// If there was no LastScrollView, but there is a superScrollView,
+		// set the LastScrollView to be the superScrollView
+		else if (superScrollView is not null)
+		{
+			LastScrollView = superScrollView;
+			StartingContentInsets = superScrollView.ContentInset;
+			StartingContentOffset = superScrollView.ContentOffset;
+
+			if (OperatingSystem.IsIOSVersionAtLeast(11, 1))
+				StartingScrollIndicatorInsets = superScrollView.VerticalScrollIndicatorInsets;
+			else
+				StartingScrollIndicatorInsets = superScrollView.ScrollIndicatorInsets;
+		}
+
+		// Calculate the move for the ScrollViews
+		if (LastScrollView is not null)
```

Comment:
This if is the exact same check that in line 312, can't they be grouped?

Reply:
No these cannot be grouped because LastScrollView and superScrollView can change before we get here after 342